### DISCUSSION TITLE
Failover

### DISCUSSION THREAD

**Discussion by AEncinas19** [strimzi/strimzi-kafka-operator#7927] [Q&A] [answered]
Hello everyone.
Is there any way to provide failover in replication? How it works?

Thanks in advance.

**Comment by scholzj** [comment, marked as answer]
I think you need to explain what do you mean with _failover in replication_. Replication is between replicas within the same Kafka cluster. When one of the brokers goes down, the client should just reconnect to the new leader. Nothing special is needed. The details how it works are more Kafka specific then Strimzi specific -> but I think the Kafka Definitive Guide book would for example describe it in detail. Is that what you meant?

**Reply by AEncinas19** [reply]
Yes, thank you very much.
Just one thing: can you access to a specific broker to see the messages saved in that broker or you only can see that all is replicated and partitioned with the describe command?
I think that with the bootstrap-server you can access always to the data regardless of where it is stored  and if a replica is working.

**Reply by scholzj** [reply]
Again, I'm not sure I entirely follow the question. Kafka has its own discovery mechanism which the clients follow.
